src/Groups/sub.jl

```diff
@@ -394,11 +394,11 @@ An exception is thrown if $G$ is not a $p$-group.
 ```jldoctest
 julia> jennings_series(dihedral_group(16))
 5-element Vector{SubPcGroup}:
- Sub-pc group of order 16
- Sub-pc group of order 4
- Sub-pc group of order 2
- Sub-pc group of order 2
- Sub-pc group of order 1
+ Sub-pc group of order 16 with 4 generators f1, f2, f3, f4
+ Sub-pc group of order 4 with 2 generators f3, f4
+ Sub-pc group of order 2 with 1 generator f4
+ Sub-pc group of order 2 with 1 generator f4
+ Sub-pc group of order 1 with 1 generator <identity> of ...
```

Comment:
Yeah this looks weird, maybe it should be
```suggestion
 Sub-pc group of order 1 with 1 generator
```


Or maybe
```suggestion
 Sub-pc group of order 1 with 1 generator f1^0
```
or
```suggestion
 Sub-pc group of order 1 with 1 generator 1
```

Reply:
The idea of omitting the identity does not work if there are multiple generators, of which some but not all are the identity.

Some further ideas on how to print the identity (I am eager to hear more):
1. `f1^0` if `f1` is a generator of the ambient group
    - I'd just always take whatever happens to be the first generator
    - it might read a bit weird to see a list of generators `f1^0, f2`
1. `_^0` or `$^0` or something artificial like that
   - avoids referencing a specific generator symbol
   - but over
1. `1` -- can be confused with ring elements 1...
1. `id` -- though one could also have a generator called `id`...
1. `identity`
1. `one`

Any other (hopefully better) ideas?

Reply:
@ThomasBreuer @lgoettgens any opinions?

Reply:
I would prefer options 3, 4 or 5

Reply:
although not asked, I would prefer 3

Reply:
While `Sub-pc group of order 1 with 1 generator 1` reads a bit weirdly I'd be OK with using that (I could also special case this case... I guess "order 1 with N generator(s)" already implies that all generators are 1 and thus we don't need to print them. And when some generators are 1 and some aren't it is clearer, e.g. `Sub-pc group of order 4 with 3 generator g1, 1, g3`

Anyway, this bit will be easy enough to change so I am happy to go with `1` for now. It's the other bits that I need to get sorted out (but I feel there is way now)